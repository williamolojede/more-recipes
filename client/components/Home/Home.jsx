import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import scrollUp from '../../utils/scrollUp';

import { fetchTopRecipes } from '../../actions/fetchRecipe';

import TopRatedRecipeList from '../shared/TopRatedRecipeList';
import Preloader from '../shared/Preloader';
import ConnectedSiteNav from '../shared/SiteNav';
import SiteFooter from '../shared/SiteFooter';
import Pagination from '../shared/Pagination';
import Search from '../shared/Search';

export class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      recipes: [],
      pages: [],
      currentPage: 1,
      limit: 6,
      isSearchActive: false
    };
  }

  componentDidMount() {
    this.props.dispatch(fetchTopRecipes(this.state.currentPage, this.state.limit));
  }

  componentWillReceiveProps(newProps) {
    if (newProps.recipes) {
      const {
        recipes,
        metaData
      } = newProps;
      this.setState({
        recipes,
        pages: metaData.pages
      });
    }
  }
  componentDidUpdate(prevProps, prevState) {
    if (prevState.currentPage !== this.state.currentPage) {
      this.props.dispatch(fetchTopRecipes(this.state.currentPage, this.state.limit));
    }
  }
  getRecipesForPage = (pageNumber) => {
    if (this.state.currentPage === pageNumber) return;
    if (pageNumber < 1 || pageNumber > this.state.pages[this.state.pages.length - 1]) {
      return;
    }
    scrollUp('16.66', '50');
    this.setState({
      currentPage: pageNumber
    });
  }

  toggleSearch =() => {
    this.setState({
      isSearchActive: !this.state.isSearchActive
    });
  }

  renderBody = () => {
    const { isFetching } = this.props;

    if (isFetching) {
      return (<Preloader />);
    } else if (!isFetching && this.state.recipes.length === 0) {
      return (
        <p className="no-recipes">
          Sorry no recipes present at the moment
          <span role="img" aria-label="sad">😢  </span>, please try again later
          <span role="img" aria-label="please">🙏</span>
        </p>
      );
    }
    return (<TopRatedRecipeList recipes={this.state.recipes} />);
  }

  render() {
    return (
      <div className="page page__home">
        <header className="site-header">
          <ConnectedSiteNav user={this.props.user} />
          <section className="site-header__hero">
            <h2>Delicious recipes, just a search away...</h2>
            <input
              className="site-header__hero--input"
              type="text"
              onFocus={this.toggleSearch}
              placeholder="what would you like to cook?"
            />
          </section>
        </header>
        <main>
          <div className="container">
            <section className="page__home--top-rated">
              <header>
                <h1>Top Rated</h1>
              </header>
              {
                this.renderBody()
              }
              {
                !this.props.isFetching ?
                  <footer>
                    <Pagination
                      currentPage={this.state.currentPage}
                      getRecipesForPage={this.getRecipesForPage}
                      pages={this.state.pages}
                    />
                  </footer> : null
              }

            </section>
          </div>
        </main>
        {
          this.state.isSearchActive && <Search toggleSearch={this.toggleSearch} />
        }
        <SiteFooter />
      </div>
    );
  }
}

Home.propTypes = {
  dispatch: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    fullname: PropTypes.string.isRequired,
  }).isRequired
};

const mapStateToProps = ({ topRecipes, isFetching, auth }) => ({
  recipes: topRecipes.recipes,
  metaData: topRecipes.metaData,
  user: auth.user,
  isFetching
});


export default connect(mapStateToProps)(Home);
