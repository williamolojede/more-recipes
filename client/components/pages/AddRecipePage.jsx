import React from 'react';

import ManageRecipes from '../shared/ManageRecipes';
import AddRecipeForm from '../shared/AddRecipeForm';

const AddRecipePage = props => (
  <ManageRecipes>
    <AddRecipeForm isEditMode={false} {...props} />
  </ManageRecipes>
);
export default AddRecipePage;
