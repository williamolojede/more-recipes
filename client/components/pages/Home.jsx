import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ReactPaginate from 'react-paginate';

import { fetchTopRecipes } from '../../actions/recipe';

import TopRatedRecipeList from '../shared/TopRatedRecipeList';
import Preloader from '../shared/Preloader';
import ConnectedSiteNav from '../shared/SiteNav';
import SiteFooter from '../shared/SiteFooter';
import Search from '../shared/Search';

import { currentUserPropTypes } from '../../config/proptypes';

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      recipes: [],
      currentPage: 1,
      limit: 6,
      last: 1,
      isSearchActive: false
    };
  }

  componentDidMount() {
    this.props.dispatch(fetchTopRecipes(this.state.currentPage, this.state.limit));
  }

  componentWillReceiveProps(newProps) {
    if (newProps.recipes) {
      const {
        recipes,
        pagination
      } = newProps;
      this.setState({
        recipes,
        last: pagination.last
      });
    }
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.currentPage !== this.state.currentPage) {
      this.props.dispatch(fetchTopRecipes(this.state.currentPage, this.state.limit));
    }
  }

  toggleSearch =() => {
    this.setState({
      isSearchActive: !this.state.isSearchActive
    });
  }

  handlePageClick = ({ selected }) => {
    this.setState({
      currentPage: selected + 1
    });
  }

  renderBody = () => {
    const { isFetching } = this.props;

    if (isFetching) {
      return (<Preloader />);
    } else if (!isFetching && this.state.recipes.length === 0) {
      return (
        <p className="no-recipes">
          Sorry no recipes present at the moment
          <span role="img" aria-label="sad">üò¢  </span>, please try again later
          <span role="img" aria-label="please">üôè</span>
        </p>
      );
    }
    return (<TopRatedRecipeList recipes={this.state.recipes} />);
  }

  render() {
    return (
      <div className="page page__home">
        <header className="site-header">
          <ConnectedSiteNav currentUser={this.props.currentUser} />
          <section className="site-header__hero">
            <h2>Delicious recipes, just a search away...</h2>
            <input
              className="site-header__hero--input"
              type="text"
              onFocus={this.toggleSearch}
              placeholder="what would you like to cook?"
            />
          </section>
        </header>
        <main>
          <div className="container">
            <section className="page__home--top-rated">
              <header>
                <h1>Top Rated</h1>
              </header>
              {
                this.renderBody()
              }
              <footer>
                <ReactPaginate
                  previousLabel={<i className="material-icons">chevron_left</i>}
                  nextLabel={<i className="material-icons">chevron_right</i>}
                  pageCount={this.state.last}
                  marginPagesDisplayed={2}
                  pageClassName="waves-effect"
                  pageRangeDisplayed={4}
                  onPageChange={this.handlePageClick}
                  activeClassName="active"
                  containerClassName="pagination"
                />
              </footer>


            </section>
          </div>
        </main>
        {
          this.state.isSearchActive && <Search toggleSearch={this.toggleSearch} />
        }
        <SiteFooter />
      </div>
    );
  }
}

Home.propTypes = {
  dispatch: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,

  ...currentUserPropTypes
};

const mapStateToProps = ({
  topRecipes: { recipes, pagination },
  isFetching,
  auth: { currentUser }
}) => ({
  recipes,
  pagination,
  currentUser,
  isFetching
});

export { Home as PureHome };
export default connect(mapStateToProps)(Home);
