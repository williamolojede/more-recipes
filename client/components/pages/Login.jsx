import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect, Link } from 'react-router-dom';

import { authPropTypes } from '../../config/proptypes';

import Footer from '../shared/Footer.jsx';
import AuthForm from '../shared/AuthForm.jsx';
import Preloader from '../shared/Preloader.jsx';
import ErrorDisplay from '../shared/ErrorDisplay.jsx';

import loginUser from '../../actions/loginUser';
import { removeAuthErrorMessage } from '../../actions/auth';


class Login extends Component {
  componentWillMount() {
    if (this.props.errorMessage) {
      this.props.dispatch(removeAuthErrorMessage());
    }
  }

  /**
   * @param {object} formData -
   * @returns {undefined} - return nothing
   */
  userLogin = (formData) => {
    this.props.dispatch(loginUser(formData));
  }

  render() {
    const {
      isFetching,
      errorMessage,
      isAuthenticated,
      location
    } = this.props;
    const { from } = location.state || { from: { pathname: '/' } };
    if (isAuthenticated) {
      return (
        <Redirect to={from} />
      );
    }

    return (
      <div className="page page__login">
        <header className="header center">
          <a href="/" className="white-text brand-logo">MoreRecipes</a>
        </header>

        <main>
          <div className="container">
            <div className="row">
              <div className="login-card card col s11 m6">
                <div className="card-content">
                  {
                    errorMessage ? <ErrorDisplay message={errorMessage} /> : null
                  }
                  <h1 className="card-title center">Log in to MoreRecipes</h1>
                  <AuthForm type="login" authFormSubmit={this.userLogin} />
                </div>
              </div>
            </div>
            <p className="white-text center">
            Donâ€™t have an account?
              <Link to="/signup" className="white-text"> Signup now</Link>
            </p>
            {
              isFetching ? <Preloader /> : null
            }
          </div>
        </main>

        <Footer />
      </div>
    );
  }
}

Login.defaultProps = {
  // the error message should not be fined by default
  errorMessage: undefined
};

Login.propTypes = authPropTypes;

export const mapStateToProps = ({ auth, isFetching }) => {
  const { isAuthenticated, errorMessage } = auth;
  return {
    isAuthenticated,
    errorMessage,
    isFetching
  };
};

export { Login as PureLogin };
export default connect(mapStateToProps)(Login);
