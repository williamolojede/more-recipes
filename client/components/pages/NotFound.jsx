import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
  <div className="page page__not-found">
    <main>
      <h1>404</h1>
      <p>Sorry, the web page your are looking for is missing
        <span role="img" aria-label="cry">ðŸ˜­</span>
      </p>
      <Link to="/">Go back to the home page</Link>
    </main>
  </div>
);


export default NotFound;
