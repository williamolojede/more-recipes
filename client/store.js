import { createStore, compose, applyMiddleware } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import { enableBatching } from 'redux-batched-actions';
import createHistory from 'history/createBrowserHistory';
import thunk from 'redux-thunk';

import rootReducer from './reducers/index';

const history = createHistory();
// Build the middleware for intercepting and dispatching navigation actions
const middleware = routerMiddleware(history);

const store = createStore(
  enableBatching(rootReducer), // to enable calling dispacth with multiple actions
  compose(
    applyMiddleware(
      thunk,
      middleware
    ),
    window.devToolsExtension ? window.devToolsExtension() : f => f
  )
);

// const store = createStore(rootReducer);

export default store;
