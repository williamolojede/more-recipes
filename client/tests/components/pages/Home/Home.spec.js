import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../../../store';
import { PureHome } from '../../../../components/pages/Home';
import user from '../../../__mocks__/user.mock';
import recipesMock from '../../../__mocks__/recipes.mock';


function setup() {
  const props = {
    dispatch: jest.fn(),
    isFetching: false,
    currentUser: user
  };
  const shallowWrapper = shallow(<PureHome {...props} />);
  const mountedWrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <PureHome {...props} />
      </BrowserRouter>
    </Provider>
  );

  return {
    shallowWrapper,
    mountedWrapper,
    props
  };
}

describe('Home', () => {
  it('should render with right amount of elements', () => {
    const { mountedWrapper } = setup();
    expect(mountedWrapper).toMatchSnapshot();
  });

  it('should render top recipes when recipes props is received', () => {
    const { shallowWrapper } = setup();
    expect(shallowWrapper).toMatchSnapshot();
    shallowWrapper.setProps({
      recipes: recipesMock.recipes,
      pagination: recipesMock.pagination
    });
    expect(shallowWrapper).toMatchSnapshot();
    expect(shallowWrapper.instance().state.recipes.length).toBe(6);
  });

  it('should render a preloader', () => {
    const { props } = setup();
    const wrapper = shallow(<PureHome {...{ ...props, ...{ isFetching: true } }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a search modal when search input is focused', () => {
    const { shallowWrapper } = setup();
    const searchInput = shallowWrapper.find('.site-header__hero--input');

    searchInput.simulate('focus');
    expect(shallowWrapper.instance().state.isSearchActive).toBe(true);
  });

  describe('handlePageClick', () => {
    it('should update currentpage state when called', () => {
      const { shallowWrapper } = setup();
      const handlePageClickSpy = jest.spyOn(shallowWrapper.instance(), 'handlePageClick');
      shallowWrapper.instance().handlePageClick({ selected: 1 });
      expect(shallowWrapper.instance().state.currentPage).toBe(2);
      expect(handlePageClickSpy).toHaveBeenCalledTimes(1);
    });
  });
});
