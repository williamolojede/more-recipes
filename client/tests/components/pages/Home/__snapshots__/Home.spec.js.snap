// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render a preloader 1`] = `
<div
  className="page page__home"
>
  <header
    className="site-header"
  >
    <Connect(SiteNav)
      currentUser={
        Object {
          "fullname": "John Doe",
          "id": 1,
        }
      }
    />
    <section
      className="site-header__hero"
    >
      <h2>
        Delicious recipes, just a search away...
      </h2>
      <input
        className="site-header__hero--input"
        onFocus={[Function]}
        placeholder="what would you like to cook?"
        type="text"
      />
    </section>
  </header>
  <main>
    <div
      className="container"
    >
      <section
        className="page__home--top-rated"
      >
        <header>
          <h1>
            Top Rated
          </h1>
        </header>
        <Preloader />
      </section>
    </div>
  </main>
  <SiteFooter />
</div>
`;

exports[`Home should render top recipes when recipes props is received 1`] = `
<div
  className="page page__home"
>
  <header
    className="site-header"
  >
    <Connect(SiteNav)
      currentUser={
        Object {
          "fullname": "John Doe",
          "id": 1,
        }
      }
    />
    <section
      className="site-header__hero"
    >
      <h2>
        Delicious recipes, just a search away...
      </h2>
      <input
        className="site-header__hero--input"
        onFocus={[Function]}
        placeholder="what would you like to cook?"
        type="text"
      />
    </section>
  </header>
  <main>
    <div
      className="container"
    >
      <section
        className="page__home--top-rated"
      >
        <header>
          <h1>
            Top Rated
          </h1>
        </header>
        <p
          className="no-recipes"
        >
          Sorry no recipes present at the moment
          <span
            aria-label="sad"
            role="img"
          >
            üò¢  
          </span>
          , please try again later
          <span
            aria-label="please"
            role="img"
          >
            üôè
          </span>
        </p>
        <footer>
          <Pagination
            currentPage={1}
            getRecipesForPage={[Function]}
            pages={Array []}
          />
        </footer>
      </section>
    </div>
  </main>
  <SiteFooter />
</div>
`;

exports[`Home should render top recipes when recipes props is received 2`] = `
<div
  className="page page__home"
>
  <header
    className="site-header"
  >
    <Connect(SiteNav)
      currentUser={
        Object {
          "fullname": "John Doe",
          "id": 1,
        }
      }
    />
    <section
      className="site-header__hero"
    >
      <h2>
        Delicious recipes, just a search away...
      </h2>
      <input
        className="site-header__hero--input"
        onFocus={[Function]}
        placeholder="what would you like to cook?"
        type="text"
      />
    </section>
  </header>
  <main>
    <div
      className="container"
    >
      <section
        className="page__home--top-rated"
      >
        <header>
          <h1>
            Top Rated
          </h1>
        </header>
        <TopRatedRecipeList
          recipes={
            Array [
              Object {
                "User": Object {
                  "fullname": "John Bull",
                  "id": 8,
                  "username": null,
                },
                "createdAt": "2017-11-03T09:51:16.180Z",
                "description": "This is one of my 'go to' recipes and it takes only about 15 minutes from start to finish. The only prep work really is mashing the garlic. The sauce is great on any pasta, but I prefer angel hair or spaghetti.",
                "downVoteCount": 0,
                "favoriteCount": 0,
                "id": 3,
                "img_url": "https://firebasestorage.googleapis.com/v0/b/more-recipes.appspot.com/o/recipes%2F1509702436404?alt=media&token=da3a7ac8-58d3-46da-a71d-c2c1c0cbdfeb",
                "ingredients": Array [
                  "2 tablespoons olive oil",
                  "3 cloves garlic",
                  "2 tablespoons capers",
                  "1 tablespoon anchovy paste",
                  "1 teaspoon red pepper flakes",
                  "1 (24 ounce) jar tomato pasta sauce",
                  "1/4 cup water",
                  "1 cup good-quality crisp white wine",
                  "2 (6.5 ounce) cans chopped clams, drained with juice reserved",
                  "salt and pepper to taste",
                  "1 pound dry spaghetti",
                  "1/4 cup freshly grated Parmesan cheese",
                  "Chopped fresh basil leaves or parsley",
                ],
                "instructions": Array [
                  "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
                  "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
                  "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
                ],
                "name": "Chef John's Spaghetti with Red Clam Sauce",
                "owner": 8,
                "reviews": Array [
                  Object {
                    "User": Object {
                      "fullname": "John Bull",
                      "id": 8,
                      "username": null,
                    },
                    "content": "gwbvgfda",
                    "createdAt": "2017-11-04T07:28:31.732Z",
                    "id": 5,
                    "recipeId": 3,
                    "updatedAt": "2017-11-04T07:28:31.732Z",
                    "userId": 8,
                  },
                ],
                "upVoteCount": 1,
                "updatedAt": "2017-11-09T11:36:49.091Z",
                "viewCount": 1,
              },
              Object {
                "User": Object {
                  "fullname": "Email Mail",
                  "id": 11,
                  "username": null,
                },
                "createdAt": "2017-11-11T09:07:36.283Z",
                "description": "This is one of my 'go to' recipes ",
                "downVoteCount": 0,
                "favoriteCount": 0,
                "id": 106,
                "img_url": "http://images.media-allrecipes.com/userphotos/720x405/4478567.jpg",
                "ingredients": Array [
                  "2 tablespoons olive oil",
                  "3 cloves garlic",
                  "2 tablespoons capers",
                  "1 tablespoon anchovy paste",
                  "1 teaspoon red pepper flakes",
                  "1 (24 ounce) jar tomato pasta sauce",
                  "1/4 cup water",
                  "1 cup good-quality crisp white wine",
                  "2 (6.5 ounce) cans chopped clams, drained with juice reserved",
                  "salt and pepper to taste",
                  "1 pound dry spaghetti",
                  "1/4 cup freshly grated Parmesan cheese",
                  "Chopped fresh basil leaves or parsley",
                ],
                "instructions": Array [
                  "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
                  "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
                  "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
                ],
                "name": "Chef John's Spaghetti with Red Clam Sauce dfbefadb nifek fedckltefdkletfdmlefdleff eifdanlkref",
                "owner": 11,
                "reviews": Array [],
                "upVoteCount": 0,
                "updatedAt": "2017-11-11T09:07:36.283Z",
                "viewCount": 0,
              },
              Object {
                "User": Object {
                  "fullname": "Email Mail",
                  "id": 11,
                  "username": null,
                },
                "createdAt": "2017-11-11T09:07:34.864Z",
                "description": "This is one of my 'go to' recipes ",
                "downVoteCount": 0,
                "favoriteCount": 0,
                "id": 101,
                "img_url": "http://images.media-allrecipes.com/userphotos/720x405/4478567.jpg",
                "ingredients": Array [
                  "2 tablespoons olive oil",
                  "3 cloves garlic",
                  "2 tablespoons capers",
                  "1 tablespoon anchovy paste",
                  "1 teaspoon red pepper flakes",
                  "1 (24 ounce) jar tomato pasta sauce",
                  "1/4 cup water",
                  "1 cup good-quality crisp white wine",
                  "2 (6.5 ounce) cans chopped clams, drained with juice reserved",
                  "salt and pepper to taste",
                  "1 pound dry spaghetti",
                  "1/4 cup freshly grated Parmesan cheese",
                  "Chopped fresh basil leaves or parsley",
                ],
                "instructions": Array [
                  "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
                  "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
                  "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
                ],
                "name": "Chef John's Spaghetti with Red Clam Sauce dfbefadb nifek fedckltefdkletfdmlefdleff eifdanlkref",
                "owner": 11,
                "reviews": Array [],
                "upVoteCount": 0,
                "updatedAt": "2017-11-11T09:07:34.864Z",
                "viewCount": 0,
              },
              Object {
                "User": Object {
                  "fullname": "Email Mail",
                  "id": 11,
                  "username": null,
                },
                "createdAt": "2017-11-11T09:07:09.090Z",
                "description": "This is one of my 'go to' recipes ",
                "downVoteCount": 0,
                "favoriteCount": 0,
                "id": 20,
                "img_url": "http://images.media-allrecipes.com/userphotos/720x405/4478567.jpg",
                "ingredients": Array [
                  "2 tablespoons olive oil",
                  "3 cloves garlic",
                  "2 tablespoons capers",
                  "1 tablespoon anchovy paste",
                  "1 teaspoon red pepper flakes",
                  "1 (24 ounce) jar tomato pasta sauce",
                  "1/4 cup water",
                  "1 cup good-quality crisp white wine",
                  "2 (6.5 ounce) cans chopped clams, drained with juice reserved",
                  "salt and pepper to taste",
                  "1 pound dry spaghetti",
                  "1/4 cup freshly grated Parmesan cheese",
                  "Chopped fresh basil leaves or parsley",
                ],
                "instructions": Array [
                  "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
                  "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
                  "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
                ],
                "name": "Chef John's Spaghetti with Red Clam Sauce dfbefadb nifek fedckltefdkletfdmlefdleff eifdanlkref",
                "owner": 11,
                "reviews": Array [],
                "upVoteCount": 0,
                "updatedAt": "2017-11-11T09:07:09.090Z",
                "viewCount": 0,
              },
              Object {
                "User": Object {
                  "fullname": "Email Mail",
                  "id": 11,
                  "username": null,
                },
                "createdAt": "2017-11-11T09:07:28.891Z",
                "description": "This is one of my 'go to' recipes ",
                "downVoteCount": 0,
                "favoriteCount": 0,
                "id": 82,
                "img_url": "http://images.media-allrecipes.com/userphotos/720x405/4478567.jpg",
                "ingredients": Array [
                  "2 tablespoons olive oil",
                  "3 cloves garlic",
                  "2 tablespoons capers",
                  "1 tablespoon anchovy paste",
                  "1 teaspoon red pepper flakes",
                  "1 (24 ounce) jar tomato pasta sauce",
                  "1/4 cup water",
                  "1 cup good-quality crisp white wine",
                  "2 (6.5 ounce) cans chopped clams, drained with juice reserved",
                  "salt and pepper to taste",
                  "1 pound dry spaghetti",
                  "1/4 cup freshly grated Parmesan cheese",
                  "Chopped fresh basil leaves or parsley",
                ],
                "instructions": Array [
                  "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
                  "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
                  "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
                ],
                "name": "Chef John's Spaghetti with Red Clam Sauce dfbefadb nifek fedckltefdkletfdmlefdleff eifdanlkref",
                "owner": 11,
                "reviews": Array [],
                "upVoteCount": 0,
                "updatedAt": "2017-11-11T09:07:28.891Z",
                "viewCount": 0,
              },
              Object {
                "User": Object {
                  "fullname": "Email Mail",
                  "id": 11,
                  "username": null,
                },
                "createdAt": "2017-11-11T09:07:11.059Z",
                "description": "This is one of my 'go to' recipes ",
                "downVoteCount": 0,
                "favoriteCount": 0,
                "id": 25,
                "img_url": "http://images.media-allrecipes.com/userphotos/720x405/4478567.jpg",
                "ingredients": Array [
                  "2 tablespoons olive oil",
                  "3 cloves garlic",
                  "2 tablespoons capers",
                  "1 tablespoon anchovy paste",
                  "1 teaspoon red pepper flakes",
                  "1 (24 ounce) jar tomato pasta sauce",
                  "1/4 cup water",
                  "1 cup good-quality crisp white wine",
                  "2 (6.5 ounce) cans chopped clams, drained with juice reserved",
                  "salt and pepper to taste",
                  "1 pound dry spaghetti",
                  "1/4 cup freshly grated Parmesan cheese",
                  "Chopped fresh basil leaves or parsley",
                ],
                "instructions": Array [
                  "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
                  "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
                  "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
                ],
                "name": "Chef John's Spaghetti with Red Clam Sauce dfbefadb nifek fedckltefdkletfdmlefdleff eifdanlkref",
                "owner": 11,
                "reviews": Array [],
                "upVoteCount": 0,
                "updatedAt": "2017-11-11T09:07:11.059Z",
                "viewCount": 0,
              },
            ]
          }
        />
        <footer>
          <Pagination
            currentPage={1}
            getRecipesForPage={[Function]}
            pages={
              Array [
                1,
                2,
                3,
              ]
            }
          />
        </footer>
      </section>
    </div>
  </main>
  <SiteFooter />
</div>
`;

exports[`Home should render with right amount of elements 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Home
        currentUser={
          Object {
            "fullname": "John Doe",
            "id": 1,
          }
        }
        dispatch={[Function]}
        isFetching={false}
      >
        <div
          className="page page__home"
        >
          <header
            className="site-header"
          >
            <Connect(SiteNav)
              currentUser={
                Object {
                  "fullname": "John Doe",
                  "id": 1,
                }
              }
            >
              <SiteNav
                currentUser={
                  Object {
                    "fullname": "John Doe",
                    "id": 1,
                  }
                }
                dispatch={[Function]}
              >
                <nav
                  className="site-header__nav"
                >
                  <div
                    className="nav-wrapper container"
                  >
                    <Link
                      className="brand-logo"
                      replace={false}
                      to="/"
                    >
                      <a
                        className="brand-logo"
                        href="/"
                        onClick={[Function]}
                      >
                        MoreRecipes
                      </a>
                    </Link>
                    <ul
                      className="right hide-on-med-and-down"
                    >
                      <li>
                        <Link
                          replace={false}
                          to="/user/1"
                        >
                          <a
                            href="/user/1"
                            onClick={[Function]}
                          >
                            Profile
                          </a>
                        </Link>
                      </li>
                      <li>
                        <Link
                          className="logout__button"
                          onClick={[Function]}
                          replace={false}
                          to="/logout"
                        >
                          <a
                            className="logout__button"
                            href="/logout"
                            onClick={[Function]}
                          >
                            Sign Out
                          </a>
                        </Link>
                      </li>
                    </ul>
                    <ul
                      className="side-nav"
                      id="mobile-nav"
                    >
                      <li>
                        <Link
                          replace={false}
                          to="/user/1"
                        >
                          <a
                            href="/user/1"
                            onClick={[Function]}
                          >
                            Profile
                          </a>
                        </Link>
                      </li>
                      <li
                        className="logout__button"
                      >
                        <Link
                          className="logout__button"
                          onClick={[Function]}
                          replace={false}
                          to="/logout"
                        >
                          <a
                            className="logout__button"
                            href="/logout"
                            onClick={[Function]}
                          >
                            Sign Out
                          </a>
                        </Link>
                      </li>
                    </ul>
                    <button
                      className="button-collapse"
                      data-activates="mobile-nav"
                    >
                      <i
                        className="material-icons"
                      >
                        menu
                      </i>
                    </button>
                  </div>
                </nav>
              </SiteNav>
            </Connect(SiteNav)>
            <section
              className="site-header__hero"
            >
              <h2>
                Delicious recipes, just a search away...
              </h2>
              <input
                className="site-header__hero--input"
                onFocus={[Function]}
                placeholder="what would you like to cook?"
                type="text"
              />
            </section>
          </header>
          <main>
            <div
              className="container"
            >
              <section
                className="page__home--top-rated"
              >
                <header>
                  <h1>
                    Top Rated
                  </h1>
                </header>
                <p
                  className="no-recipes"
                >
                  Sorry no recipes present at the moment
                  <span
                    aria-label="sad"
                    role="img"
                  >
                    üò¢  
                  </span>
                  , please try again later
                  <span
                    aria-label="please"
                    role="img"
                  >
                    üôè
                  </span>
                </p>
                <footer>
                  <Pagination
                    currentPage={1}
                    getRecipesForPage={[Function]}
                    pages={Array []}
                  >
                    <ul
                      className="pagination"
                    >
                      <li
                        className="disabled"
                        id="pagination__prev"
                      >
                        <button
                          onClick={[Function]}
                        >
                          <i
                            className="material-icons"
                          >
                            chevron_left
                          </i>
                        </button>
                      </li>
                      <li
                        className="waves-effect"
                        id="pagination__next"
                      >
                        <button
                          onClick={[Function]}
                        >
                          <i
                            className="material-icons"
                          >
                            chevron_right
                          </i>
                        </button>
                      </li>
                    </ul>
                  </Pagination>
                </footer>
              </section>
            </div>
          </main>
          <SiteFooter>
            <footer
              className="page-footer"
            >
              <div
                className="container"
              >
                <p>
                  Crafted with 
                  <i
                    className="mdi mdi-heart"
                  />
                   by 
                  <a
                    href="http://william.ng/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    William I. Olojede
                  </a>
                </p>
              </div>
            </footer>
          </SiteFooter>
        </div>
      </Home>
    </Router>
  </BrowserRouter>
</Provider>
`;
